src/hir/translator.rs

```diff
@@ -226,6 +244,16 @@ impl<'a> Visitor for TranslatorI<'a> {
                 Some(self.modifier_stack.iter().map(|&m| m.to_owned()).collect())
             };
 
+            // Add the (optional) statements which should be at the function top-level
+            if *self.with_vm_skip {
+                actions.insert(
+                    0,
+                    Hir::Statement(hir::Statement {
+                        ty: hir::SupportedStatement::VmSkip,
+                    }),
+                );
+            }
+
```

Comment:
I played a bit with this one too, what would be the more idiomatic way @alexfertel ?

Something like this?
```Rust
            // Add the (optional) statements which should be at the function top-level
            let mut additional_statements = vec![];
            if *self.with_vm_skip {
                additional_statements.push(Hir::Statement(hir::Statement {
                    ty: hir::SupportedStatement::VmSkip,
                }));
            }

            additional_statements.extend(actions);

            let hir = Hir::FunctionDefinition(hir::FunctionDefinition {
                identifier: function_name,
                ty: hir::FunctionTy::Function,
                span: condition.span,
                modifiers,
                children: Some(additional_statements)
            });
```

Reply:
I think that's even less efficient because now we allocate another `Vec`, but still copy all the elements from one to the other in `extend`, hehe.

This would need to be measured in practice though, cause it may be optimized away.

It's fine though, the performance here is negligible, it's better to keep it readable, and using `insert` is 👍